src/CPUSolver.cpp

```diff
@@ -1717,20 +1709,15 @@ void CPUSolver::computeKeff() {
     MPI_Comm comm = _geometry->getMPICart();
 
     /* Copy local rates */
-    double local_rates[3];
-    for (int i=0; i < 3; i++)
-      local_rates[i] = rates[i];
+    double local_rate[0];
+    local_rate[0] = fission[0];
 
     /* Reduce computed rates */
-    MPI_Allreduce(local_rates, rates, 3, MPI_DOUBLE, MPI_SUM, comm);
+    MPI_Allreduce(local_rate, fission, 1, MPI_DOUBLE, MPI_SUM, comm);
   }
 #endif
 
-  /* Compute k-eff from fission, absorption, and leakage rates */
-  double fission = rates[0];
-  double absorption = rates[1];
-  double leakage = rates[2];
-  _k_eff = fission / (absorption + leakage);
+  _k_eff *= fission[0] / _num_FSRs;
```

Comment:
In my opinion, it's better to keep both ways of keff calculation since OpenMOC is a code for research. Options could be provided to users as first, second or both, and the default value is set to be the fission rate one.